pdr_backend/predictoor/predictoor_agent.py

```diff
@@ -118,6 +120,26 @@ def take_step(self):
         self.prev_block_number = UnixTimeS(self.cur_block_number)
         self.prev_block_timestamp = UnixTimeS(self.cur_timestamp)
 
```

Comment:
Can the newly added code be put into a sub-method please? The current method is already quite complex, and the new code really hurts readability.